src/app/hooks/useClickTracker/index.jsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-console */
+import { useEffect, useRef, useContext, useCallback } from 'react';
+import { sendEventBeacon } from '#containers/ATIAnalytics/beacon/index';
+import { getComponentInfo } from '#app/lib/analyticsUtils/index';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { RequestContext } from '#contexts/RequestContext';
+import { buildATIClickParams } from '#containers/ATIAnalytics/params';
+import { isValidClick } from './clickTypes';
+
+const useClickTracker = ({ pageData, componentName, href } = {}) => {
+  let eventTrackingProps;
+
+  const requestContext = useContext(RequestContext);
+  const serviceContext = useContext(ServiceContext);
+  const { service } = serviceContext;
+  const clickRef = useRef(null);
+
+  try {
+    eventTrackingProps = buildATIClickParams(
+      pageData,
+      requestContext,
+      serviceContext,
+    );
+  } catch (error) {
+    console.error(
+      `ATI Event Tracking Error: Could not parse tracking values from page data:\n${error.message}`,
+    );
+    eventTrackingProps = null;
+  }
+
+  const handleClick = useCallback(
+    event => {
+      let componentInfo;
+      event.stopPropagation();
+      event.preventDefault();
+
+      if (isValidClick(event)) {
+        clickRef.current?.removeEventListener('click', handleClick);
+
+        try {
+          componentInfo = getComponentInfo({
+            result: href || event.target.href || window.location.href,
+            componentName,
+            componentData: {
+              actionLabel: 'click',
+              child: event.target.tagName,
+            },
+          });
+        } catch (error) {
+          console.error(
+            `Error getting component info for ATI Tracking: ${error.message}`,
+          );
+          componentInfo = null;
+        }
+
+        if (eventTrackingProps && componentInfo) {
+          sendEventBeacon({
+            type: 'click',
+            componentName,
+            service,
+            componentInfo,
+            ...eventTrackingProps,
+          })
+            .then(() => {
+              if (href || event.target.href) {
```

Comment:
this could be a constant e.g. `const nextPageUrl = href || event.target.href;` or something similar